doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -185,129 +166,118 @@ Newly opened modules and sections inherit the current settings.
+   Displays the name and type of objects matching :n:`@search_item`\s
+   in the selected goals (if a proof is open)
+   as well as accessible theorems, axioms, etc..
+   It's useful for finding the names of library lemmas.
 
-      where
-      :n:`@term_pattern_string` is a term_pattern, a string, or a string followed
-      by a scope delimiting key `%key`.  This generalization of ``Search`` searches
-      for all objects whose statement or type contains a subterm matching
-      :n:`@term_pattern` (or :n:`@qualid` if :n:`@string` is the notation for a reference
-      qualid) and whose name contains all string of the request that
-      correspond to valid identifiers. If a term_pattern or a string is
-      prefixed by `-`, the search excludes the objects that mention that
-      term_pattern or that string.
+   * :n:`@string` - If :n:`@string` is a substring of a valid identifier,
+     search for objects whose name contains :n:`@string`. If :n:`@string` is a notation
+     string associated with a :n:`@qualid`, that's equivalent to :cmd:`Search` :n:`@qualid`.
+     For example, specifying "+" or "_ + _", which are notations for "plus", are equivalent
+     to :cmd:`Search` :n:`plus`.
 
-   .. cmdv:: Search {+ {? -}@term_pattern_string} inside {+ @qualid }
+   * :n:`@scope` - limits the search to the scope bound to
+     the delimiting key :n:`@scope`, such as, for example, :n:`%nat`
+     (see Section :ref:`LocalInterpretationRulesForNotations`).
 
-      This restricts the search to constructions defined in the modules
-      named by the given :n:`qualid` sequence.
+   If you specify multiple :n:`@search_item`\s, all the conditions must be satisfied
+   for the object to be displayed.
 
-   .. cmdv:: Search {+ {? -}@term_pattern_string} outside {+ @qualid }
+   * :n:`@one_term` - Search for objects containing a subterm matching the pattern
+     :n:`@one_term` in which holes of the pattern are indicated by `_` or :n:`?@ident`.
+     If the same :n:`?@ident` occurs more than once in the pattern, all occurrences must
+     match the same value.
 
-      This restricts the search to constructions not defined in the modules
-      named by the given :n:`qualid` sequence.
+   * :n:`{? - }` - excludes objects that contain the :n:`@string` or :n:`@one_term`
+   * :n:`inside {+ @qualid }` - limit the search to the specified modules
+   * :n:`outside {+ @qualid }` - exclude the specified modules from the search
 
-   .. cmdv:: @selector: Search {+ {? -}@term_pattern_string}
+   .. exn:: The reference @qualid was not found in the current environment.
 
-      This specifies the goal on which to search hypothesis (see
-      Section :ref:`invocation-of-tactics`).
-      By default the 1st goal is searched. This variant can
-      be combined with other variants presented here.
+      There is no constant in the environment named :n:`@qualid`.
 
-      .. example::
+      .. example:: :cmd:`Search` examples
```

Comment:
This example is indented too much, as if it was part of the explanation of the error message above.